feature/experimental/tunnel/otg_tests/tunnel_interface_based_ipv4_gre_encapsulation_test/tunnel_interface_based_ipv4_gre_encapsulation_test.go

```diff
@@ -0,0 +1,482 @@
+}
+func verifyTrafficStatistics(t *testing.T, ate *ondatra.ATEDevice, flowName string, wantLoss bool) {
+	otg := ate.OTG()
+	t.Logf("Traffic Loss Test Validation for flow %s\n", flowName)
+	recvMetric := gnmi.Get(t, otg, gnmi.OTG().Flow(flowName).State())
+	txPackets := recvMetric.GetCounters().GetOutPkts()
+	t.Logf("Flow: %s transmitted packets: %d !", flowName, txPackets)
+	rxPackets := recvMetric.GetCounters().GetInPkts()
+	t.Logf("Flow: %s received packets: %d !", flowName, rxPackets)
+	lostPackets := txPackets - rxPackets
+	t.Logf("Flow: %s lost packets: %d !", flowName, lostPackets)
+	lossPct := lostPackets * 100 / txPackets
+	t.Logf("Flow: %s packet loss percent : %d !", flowName, lossPct)
+	if wantLoss {
+		if lossPct > uint64(tolerance) {
+			t.Errorf("Traffic Loss for Flow: %s but got %v, want 0 Failed.", flowName, lossPct)
+		} else {
+			t.Logf("No Traffic Loss Test Passed!!")
+		}
+	} else {
+		if lossPct < 100-uint64(tolerance) {
+			t.Errorf("Traffic is expected to fail but flow :%s  got %v, want 100%% Failed.", flowName, lossPct)
+		} else {
+			t.Logf("Traffic Loss Test Passed!!")
+		}
+	}
+}
+func sendTraffic(t *testing.T, ate *ondatra.ATEDevice) {
+	otg := ate.OTG()
+	t.Logf("Starting traffic")
+	otg.StartTraffic(t)
+	time.Sleep(time.Duration(trafficDuration) * time.Second)
+	t.Logf("Stop traffic")
+	otg.StopTraffic(t)
+}
+func configureOtgPorts(top gosnappi.Config, port *ondatra.Port, name string, mac string, ipv4Address string, ipv4Gateway string, ipv4Mask uint8) {
+
+	top.Ports().Add().SetName(port.ID())
+	//port1
+	iDutDev := top.Devices().Add().SetName(name)
+	iDutEth := iDutDev.Ethernets().Add().SetName(name + ".Eth").SetMac(mac)
+	iDutEth.Connection().SetChoice(gosnappi.EthernetConnectionChoice.PORT_NAME).SetPortName(port.ID()) //port.Name()
+	iDutIpv4 := iDutEth.Ipv4Addresses().Add().SetName(name + ".IPv4")
+	iDutIpv4.SetAddress(ipv4Address).SetGateway(ipv4Gateway).SetPrefix(uint32(ipv4Mask))
+
+}
+func configureTrafficFlowsToEncasulation(t *testing.T, top gosnappi.Config, port1 *ondatra.Port, port2 *ondatra.Port, port3 *ondatra.Port, peer *attrs.Attributes, destMac string) {
+	t.Logf("configure IPv4 flow from %s ", port1.Name())
+	// Set config flow
+	flow1ipv4 := top.Flows().Add().SetName("IPv4")
+	flow1ipv4.Metrics().SetEnable(true)
+	//TxRx ports
+	flow1ipv4.TxRx().Port().SetTxName(port1.ID()).SetRxNames([]string{port2.ID(), port3.ID()})
+	// Flow settings.
+	flow1ipv4.Size().SetFixed(512)
+	flow1ipv4.Rate().SetPps(trafficRatePps)
+	flow1ipv4.Duration().SetChoice("continuous")
+	// Ethernet header
+	f1e1 := flow1ipv4.Packet().Add().Ethernet()
+	f1e1.Src().SetValue(peer.MAC)
+	f1e1.Dst().SetValue(destMac)
+	// IP header
+	f1v4 := flow1ipv4.Packet().Add().Ipv4()
+	// V4 protocol
+	f1v4.Protocol().SetChoice("value").SetValue(6)
+	// V4 source
+	f1v4.Src().Increment().SetStart(peer.IPv4).SetCount(200)
+	// V4 destination
+	f1v4.Dst().Increment().SetStart(encapInnerDesIpv4Network).SetCount(200)
+	// Add L4 protocol
```

Comment:
Please remove all comments as traffic flow APIs are self-explanatory and commonly used.